docs/revive_compiler/differences_yul_translation.md

```diff
@@ -0,0 +1,98 @@
+---
+sidebar_position: 4
+---
+
+# Differences in YUL IR translation
+This section highlights some potentially observable differences in the [YUL EVM dialect](https://docs.soliditylang.org/en/latest/yul.html#evm-dialect) translation compared to Ethereum Solidity.
+
+Solidity developers deploying dApps to [`pallet-revive`](https://github.com/paritytech/polkadot-sdk/tree/master/substrate/frame/revive) ought to read and understand this section well.
+
+:::tip
+
+For a high level overview about the differences in the overall runtime architecture, please refer first to the [Differences to Ethereum](../differences_to_eth.md) chapter.
+
+:::
+
+## Deploy code vs. runtime code
+Our contract runtime does not differentiate between runtime code and deploy (constructor) code. Instead, both are emitted into a single PVM contract code blob and live on-chain. Therefor, in EVM termonology, the deploy code equals the runtime code (with the exception that in constructor code, the `codesize` can be larger, since the constructor arguments account to the code size too).
+
```

Comment:
> with the exception that in constructor code, the `codesize` can be larger, since the constructor arguments account to the code size too

codesize is an ethereum reference here, I don't fully get this sentence

Reply:
I wanted to say if you call `codesize` in the constructor you get a different value than in runtime code, because despite the code being the same, `codesize` will account for the arguments as well. I'm going to rephrase it thanks!